### ISSUE TITLE
Virtualization Error

### ISSUE THREAD

**Issue by rupom96** [KevinVandy/material-react-table#445]
### material-react-table version

v1.9.2

### react & react-dom versions

17.0.2

### Describe the bug and the steps to reproduce it

In your virtualized example, described in: "https://www.material-react-table.com/docs/guides/virtualization" page, whenever a user scrolling the table with mouse wheel a bit faster, the whole page of your website collapses and says-

"Error: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
 
A client-side exception has occurred, see here for more info: https://nextjs.org/docs/messages/client-side-exception-occurred"

And the whole grid disappears from your website, saying that its a client side exception.  

### Minimal, Reproducible Example - (Optional, but Recommended)

Just visit "https://www.material-react-table.com/docs/guides/virtualization" and scroll down your virtualized example to see the error.

### Screenshots or Videos (Optional)

https://user-images.githubusercontent.com/31258758/228140506-432c26ae-5952-425a-bda1-55cba871c736.mp4

### Do you intend to try to help solve this bug with your own PR?

None

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
I have not been able to reproduce this myself yet, but thanks for reporting

**Comment by rupom96** [comment]
@KevinVandy By the way, sorry I didn't mention earlier that I was using"Mozilla Firefox" while producing this bug. After your comment that you couldn't produce this bug, I immediately tested with chrome and found in chrome, it is working just fine.

That means this bug arises only in mozilla firefox (I am using updated version of it fyi).

**Comment by snoobism** [comment]
I am also encountering this issue on Firefox. Are there any workarounds?

**Comment by danvmanici** [comment]
@KevinVandy  I am also having this problem and I could reproduce it on Mozilla Firefox using Windows , but couldn't reproduce it on Mozilla Firefox using macOS.
Maybe it helps to track the issue.

**Comment by KevinVandy** [comment]
I think I'm going to need more community help here. For Firefox on both Mac and Windows on my personal devices, I still cannot reproduce.

**Comment by Scott-Barnett** [comment]
I can also reproduce this on Firefox for Windows. I've had a test and from what I can tell, it appears when scaling is enabled - I haven't managed to replicate it on 100% but 125% seems to be pretty repeatable.

The way I replicate it is by opening the Virtualized Example and then scrolling at a steady speed downwards. When I do this, the data refreshes slow down and stop keeping up with the scrolling and eventually the table dies. When testing with your virtualized example in development mode, I got the following error:
![image](https://user-images.githubusercontent.com/43785220/233793460-f6a15d31-99ca-4c34-8b40-526cb1ab937f.png)

**Comment by KevinVandy** [comment]
I can now confirm that whenever Firefox is not at 100% scaling, it seems to crash during virtual scrolling. It seems to happen on all OSes. This is so dumb. It's the only browser that I can tell has this problem. I might have to rewrite the entire virtualization feature with a different approach. Help would be appreciated though.

**Comment by KevinVandy** [comment]
> I'd be happy to look into this. Any ideas about which parts of the current implementation are causing the issues? It seems like TanStack Table is repeatedly applying the opposite correction to rows, making their .start-value flicker between the same two values. My understanding is that a correction is applied when the actual height of rows is determined, so is it somehow related to the measureElement function passed to useVirtualizer?
> ![image](https://user-images.githubusercontent.com/70698952/236489153-210a6c88-efaf-4262-ae5a-7fd4631b22df.png)
> 

Yes, I think the error is in TanStack Virtual itself. But have not gathered enough evidence to create an issue yet there. Would love to just create a reproducible issue on that repo first

**Comment by MarkusEllyton** [comment]
So I followed your hunch of it being an internal bug within TanStack Virtual, and tried creating a minimal reproducible example using just Virtual. The example (https://stackblitz.com/edit/react-ts-tyut32?file=App.tsx) is more or less a copy of the example from the official docs (https://tanstack.com/virtual/v3/docs/examples/react/fixed), I've just added a bottom border to each row, and that seems to trigger it along with

```ts  
ref={(node: HTMLTableRowElement) => {
      if (node) {
          rowRef.current = node;
          measureElement?.(node);
      }
  }}
 ```
 inspired by the way it's done in `MRT_TableBodyRow`. The error doesn't occur without it, not sure why, but maybe you have more insights.

**Comment by KevinVandy** [comment]
> So I followed your hunch of it being an internal bug within TanStack Virtual, and tried creating a minimal reproducible example using just Virtual. The example (https://stackblitz.com/edit/react-ts-tyut32?file=App.tsx) is more or less a copy of the example from the official docs (https://tanstack.com/virtual/v3/docs/examples/react/fixed), I've just added a bottom border to each row, and that seems to trigger it along with
> 
> ```ts  
> ref={(node) => {
>       if (node) {
>           rowRef.current = node;
>           measureElement?.(node);
>       }
>   }}
>  ```
>  inspired by the way it's done in `MRT_TableBodyRow`. The error doesn't occur without it, not sure why, but maybe you have more insights.
>  
>  EDIT: I almost forgot to mention, passing this to `useVirtualizer` also seems to be necessary
>  
>  ```ts
>     measureElement: (element) => element?.getBoundingClientRect().height,
>  ```

Awesome investigation. This is starting to make sense, because Firefox treats row borders differently than all other browsers. That's why this lib used border collapse: separate, just to fix borders not showing on rows in Firefox.

**Comment by MarkusEllyton** [comment]
I thought it made sense too, but the more I try to figure out the cause the more confused I get.

What I'm almost certain of at this point is that it's related to fractional pixels. If you zoom in at multiples of 100% such that all sizes are integers, the problem goes away. I've also found some possible solutions. 
1. Rounding `measureElement` like the default implementation from Tanstack Virtual. 
2. Using `box-sizing: border-box` on the rows to make sure the height of the rows never become fractional.  

I don't know if either of these options are viable with MRT. Oh, well. I'll keep digging.

**Comment by KevinVandy** [comment]
@MarkusEllyton @Scott-Barnett @danvmanici @snoobism @rupom96 This has potentially been fixed in v1.11.3. I'd love some feedback here for whether or not you all think it is fixed or not.

**Comment by Scott-Barnett** [comment]
I can confirm that since that update, I haven't been able to reproduce the issue so it appears to be sorted! Thank you very much for looking into it!
